sys/net/application_layer/sock_dodtls/sock_dodtls.c

```diff
@@ -205,55 +195,14 @@ static int _connect_server(const sock_udp_ep_t *server,
         }
         goto exit;
     }
-    if ((res = sock_dtls_create(&_dtls_sock, &_udp_sock, creds->tag,
-                                SOCK_DTLS_1_2, SOCK_DTLS_CLIENT)) < 0) {
-        puts("Unable to create DTLS sock\n");
-        _close_session(creds->tag, creds->type);
-        goto exit;
-    }
-
-    start = _now_ms();
-    try_start = start;
-    while (((try_start = _now_ms()) - start) < SOCK_DODTLS_SESSION_TIMEOUT_MS) {
-        memset(&_server_session, 0, sizeof(_server_session));
-        if ((res = sock_dtls_session_init(&_dtls_sock, server,
-                                          &_server_session)) >= 0) {
-            uint32_t try_duration;
-
-            res = sock_dtls_recv(&_dtls_sock, &_server_session, _dns_buf,
-                                 sizeof(_dns_buf), timeout * US_PER_MS);
-            if (res == -SOCK_DTLS_HANDSHAKE) {
-                break;
-            }
-            DEBUG("Unable to establish DTLS handshake: %d (timeout: %luus)\n",
-                  -res, (long unsigned)timeout * US_PER_MS);
-            sock_dtls_session_destroy(&_dtls_sock, &_server_session);
-            try_duration = _now_ms() - try_start;
-            if (try_duration < timeout) {
-                _sleep_ms(timeout - try_duration);
-            }
-            /* see https://datatracker.ietf.org/doc/html/rfc6347#section-4.2.4.1 */
-            timeout *= 2U;
-        }
-        else {
-            DEBUG("Unable to initialize DTLS session: %d\n", -res);
-            sock_dtls_session_destroy(&_dtls_sock, &_server_session);
-        }
-    }
-    if (res != -SOCK_DTLS_HANDSHAKE) {
-        res = -ETIMEDOUT;
-        _close_session(creds->tag, creds->type);
-        goto exit;
-    }
-    else {
-        res = 0;
-    }
 
+    res = sock_dtls_establish_session(&_udp_sock, &_dtls_sock, &_server_session,
+                                      creds->tag, &local, server, _dns_buf,
+                                      sizeof(_dns_buf));
```

Comment:
Oh and I saw in my IDE that `_dns_buf` is 128 bytes by default, but at least 160 bytes are required

Reply:
That's a tricky one. There is no standard about how large the server response might be.
Looks like for @miri64's case a 128 byte buffer was enough.

When I tested this with the Californium CoAPS implementation I needed a larger buffer for the handshake while aiocoap was also fine with 128 bytes.

Reply:
We tried to find some guide line in #18724 but there is none, the 160 where found by benpicco searching for what work in his setup

The standard says that it is a response with the same limits as any other message -> to high

Reply:
> Looks like for @miri64's case a 128 byte buffer was enough.

Yepp. gCoAP buffers are that large ;-)